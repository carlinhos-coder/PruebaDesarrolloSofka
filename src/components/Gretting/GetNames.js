import React, { useEffect, useState } from 'react';
import { Modal, ModalBody, ModalHeader, ModalFooter } from 'reactstrap';
import { deleteName, updateName } from '../../helper/fetchApi';
import "bootstrap/dist/css/bootstrap.min.css";
import Swal from 'sweetalert2';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons';


export const GetNames = ({ names, setNames, setChangeComponent, setloading }) => {

    const [modalEdit, setModalEdit] = useState(false);
    const [modalDelete, setModalDelete] = useState(false);
    const [nameSelected, setNameSelected] = useState({
        id: '',
        nombre: '',
        estado: ''
    });
    const [check, setCheck] = useState("1");
    const [tablaUsuarios, setTablaUsuarios] = useState([]);
    const [busqueda, setBusqueda] = useState("");

    const modifyStatus = (e) => {
        setCheck(e.target.value)
    }

    const selectName = (elemento, caso) => {
        setNameSelected(elemento);
        (caso === 'Editar') ? setModalEdit(true) : setModalDelete(true)
    }

    const handleChange = e => {
        const { name, value } = e.target;
        setNameSelected((prevState) => ({
            ...prevState,
            [name]: value,
        }));
    }
    const handleFilter = e => {
        setBusqueda(e.target.value);
        filtrar(e.target.value);
    }

    const filtrar = (terminoBusqueda) => {
        var resultadosBusqueda = names.filter((elemento) =>
            elemento.nombre.toLowerCase().includes(terminoBusqueda.toLowerCase())
        );
        setTablaUsuarios(resultadosBusqueda);

    }
    useEffect(() => {
        setTablaUsuarios(names);
    }, [names])

    const edit = () => {
        var dataNueva = names;
        dataNueva.forEach(name => {
            if (name.id === nameSelected.id) {
                name.nombre = nameSelected.nombre;
                if (check === "2") {
                    name.estado = false
                } else if (check === "1") {
                    name.estado = true;
                }
            }
        });
        setloading(true)
        updateName(nameSelected)
            .then(res => res.json())
            .then(data => data.status === 400 ? Swal.fire("No se actualizó correctamente") :
                setloading(false), Swal.fire("Se actualizó correctamente"))
        setModalEdit(false);

    }

    const deleteBd = () => {
        setloading(true)
        deleteName(nameSelected)
            .then(res => res.status === 400 ? Swal.fire("No se borró correctamente") :
                setloading(false), Swal.fire("Se borró correctamente")
            )
        setNames(names.filter(name => name.id !== nameSelected.id));
        setModalDelete(false);
    }

    return <div className='App'>
        <div className="containerInput">
            <input
                className="form-control inputBuscar"
                value={busqueda}
                placeholder="Búsqueda por Nombre"
                onChange={handleFilter}
            />
            <button className="btn btn-success">
                <FontAwesomeIcon icon={faSearch} />
            </button>
        </div>

        <button onClick={() => setChangeComponent(true)} >
            Volver
        </button>
        <table className="table table-sm table-bordered">
            <thead>
                <tr>
                    <th>ID</th>
                    <th>Nombre</th>
                    <th>Estado</th>
                    <th>Acciones</th>
                </tr>
            </thead>
            <tbody>
                {tablaUsuarios.map(element => (
                    <tr key={element.id}>
                        <td>{element.id}</td>
                        <td>{element.nombre}</td>
                        <td>{element.estado ? "Activo" : "Inactivo"}</td>
                        <td><button className="btn btn-primary" onClick={() => selectName(element, 'Editar')}>Editar</button> {"   "}
                            <button className="btn btn-danger" onClick={() => selectName(element, 'Eliminar')}>Eliminar</button></td>
                    </tr>
                ))
                }
            </tbody>
        </table>

        <Modal isOpen={modalEdit}>
            <ModalHeader>
                <div>
                    <h3>Editar Nombre</h3>
                </div>
            </ModalHeader>
            <ModalBody>
                <div className="form-group">
                    <label>ID</label>
                    <input
                        className="form-control"
                        readOnly
                        type="text"
                        name="id"
                        value={nameSelected && nameSelected.id}
                    />
                    <br />

                    <label>Nombre</label>
                    <input
                        className="form-control"
                        type="text"
                        name="nombre"
                        value={nameSelected && nameSelected.nombre}
                        onChange={handleChange}
                    />
                    <br />

                    <label >Estado</label>
                    <br />
                    <div className="hijo">
                        <input id="radio1" value="1" onChange={modifyStatus} checked={check === "1" ? true : false} className="form-check-input" type="radio" />
                        <label className="form-check">
                            Activo
                        </label></div>
                    <div className="hijo">
                        <input id="radio2" value="2" onChange={modifyStatus} checked={check === "2" ? true : false} className="form-check-input" type="radio" />
                        <label className="form-check">
                            Inactivo
                        </label></div>
                    <br />
                </div>
            </ModalBody>
            <ModalFooter>
                <button className="btn btn-primary" onClick={() => edit()}>
                    Actualizar
                </button>
                <button
                    className="btn btn-danger"
                    onClick={() => setModalEdit(false)}
                >
                    Cancelar
                </button>
            </ModalFooter>
        </Modal>

        <Modal isOpen={modalDelete}>
            <ModalBody>
                Estás Seguro que deseas eliminar el nombre {nameSelected && nameSelected.nombre}
            </ModalBody>
            <ModalFooter>
                <button className="btn btn-danger" onClick={() => deleteBd()}>
                    Sí
                </button>
                <button
                    className="btn btn-primary"
                    onClick={() => setModalDelete(false)}
                >
                    No
                </button>
            </ModalFooter>
        </Modal>
    </div>;
};
