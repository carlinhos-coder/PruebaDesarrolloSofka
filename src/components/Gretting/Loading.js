import React from 'react';
import { Spinner } from 'reactstrap';

export const Loading = () => {
    return <div>
        <div className='fatherLoading'>
            <div className='sonLoading'>
                <Spinner color="primary" className='spinner' />
            </div>
        </div>
    </div>;
};
