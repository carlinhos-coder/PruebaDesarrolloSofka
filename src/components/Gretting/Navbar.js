import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startLogout } from '../../actions/auth'

export const Navbar = () => {

    const dispatch = useDispatch()
    const { name } = useSelector(state => state.auth)


    const handleLogout = () => {
        dispatch(startLogout())
    }

    return (
        <div className='form__sidebar-navbar'>
            <h3 className='mt-5'>
                <i className='far fa-moon'></i>
                <span>{name}</span>
            </h3>
            <button
                onClick={handleLogout}
                className='btn'>
                LogOut
            </button>
        </div>
    )
}
