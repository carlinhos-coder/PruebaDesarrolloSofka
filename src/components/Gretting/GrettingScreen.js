import React, { useState } from 'react';
import { useForm } from '../../hooks/useForm';
import { db } from '../../firebase/firebase-config';
import Swal from 'sweetalert2';
import { getName, getNames, setName } from '../../helper/fetchApi';
import { GetNames } from './GetNames';
import { Loading } from './Loading';

export const GrettingScreen = () => {

    const [formValues, handleInputChange, reset, validateInput] = useForm();
    const [check, setCheck] = useState("1");
    const [names, setNames] = useState([])
    const [changeComponent, setChangeComponent] = useState(true)
    const [loading, setloading] = useState(false);

    if (loading) {
        return <Loading />
    }

    const alphabeticComparator = (a, b) =>
        a.id < b.id ? -1 : a.id > b.id ? 1 : 0;

    const handleGreet = async (e) => {
        e.preventDefault()
        await db.collection('saludo').doc().set(formValues)
        validateName(formValues.nombre, idiomasSaludo)
        reset()
    }

    const handleAddBD = async (e) => {
        e.preventDefault()
        saveName(formValues.nombre)
        reset()
    }

    const validateName = (name, language) => {
        setloading(true)
        getName(name)
            .then(res =>
                res ? (setloading(false), Swal.fire(language[check - 1] + ' ' + name)) :
                    (Swal.fire("No existes en la base de datos"), setloading(false))
            ).catch(() => Error)
    }

    const saveName = (name) => {
        setloading(true)
        setName(name)
            .then(res =>
                res ? (setloading(false), Swal.fire("Se guardó correctamente")) :
                    Swal.fire("No se guardó correctamente")
            ).catch(() => Error)
    }

    const handleDespedir = (e) => {
        e.preventDefault();
        validateName(formValues.nombre, idiomasDespedida)
        reset()
    }

    const handleName = (e) => {
        e.preventDefault();
        validateName(formValues.nombre, idiomasNombre)
        reset()
    }
    const changeStatus = (e) => {
        setCheck(e.target.value)
    }

    const handleGetNames = (e) => {
        e.preventDefault();
        setloading(true)
        getNames()
            .then(resp => resp.json())
            .then(data => {
                setNames(data.sort(alphabeticComparator))
                setloading(false)
            });
        setChangeComponent(false)


    }
    const idiomasSaludo = [
        "Hello:", "Hola:"
    ]

    const idiomasDespedida = [
        "See you later:", "Hasta luego:"
    ]

    const idiomasNombre = [
        "My name is:", "Mi nombre es:"
    ]
    return <div>
        {changeComponent &&
            <div className="padre2">
                <div className="hijo2">
                    <div className='abs-center'>

                        <form>
                            <h1 className='form_title'>Acá te saludamos por tu nombre</h1>

                            <label className='labels'>Nombre </label>

                            <input
                                type="text"
                                name="nombre"
                                className="mt-10 input"
                                autoComplete="off"
                                onChange={handleInputChange}
                                value={formValues?.nombre ? formValues?.nombre : ""}

                            />
                            {validateInput &&
                                <div className='auth__alert-error'>
                                    Debes ingresar nombre
                                </div>
                            }



                            <div className="padre">
                                <div className="hijo">
                                    <input id="radio1" value="1" onChange={changeStatus} checked={check === "1" ? true : false} className="form-check-input" type="radio" />
                                    <label className="form-check">
                                        Inglés
                                    </label></div>
                                <div className="hijo">
                                    <input id="radio2" value="2" onChange={changeStatus} checked={check === "2" ? true : false} className="form-check-input" type="radio" />
                                    <label className="form-check">
                                        Español
                                    </label></div>
                            </div>

                            <div className="padre">
                                <div className="hijo"><button disabled={validateInput} onClick={handleGreet}
                                    className='btns btn-primary btn-block buttom'>Saludar</button></div>
                                <div className="hijo"><button disabled={validateInput} onClick={handleName} className='btns btn-primary btn-block buttom'>Nombre</button></div>
                                <div className="hijo"><button disabled={validateInput} onClick={handleDespedir} className='btns btn-primary btn-block buttom'>Despedir</button></div>
                                <div className="hijo"><button disabled={validateInput} onClick={handleAddBD} className='btns btn-primary btn-block buttom'>Guardar</button></div>
                                <div className="hijo"><button onClick={handleGetNames} className='btns btn-primary btn-block buttom'>Consultar</button></div>
                            </div>
                        </form>
                    </div>
                </div>
            </div>
        }
        {!changeComponent &&
            <GetNames names={names} setNames={setNames} setChangeComponent={setChangeComponent} setloading={setloading} />
        }

    </div >
};

