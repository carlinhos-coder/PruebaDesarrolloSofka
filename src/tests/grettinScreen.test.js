
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import { GrettingScreen } from '../components/Gretting/GrettingScreen'

configure({ adapter: new Adapter() });
const wrapper = shallow(<GrettingScreen />)
describe('proof <GrettingScreen/>', () => {
    test('should show the component correctly ', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('should send name when clic in saludar ', () => {
        wrapper.find('form').simulate('click', { preventDefault() { } })
    });



})