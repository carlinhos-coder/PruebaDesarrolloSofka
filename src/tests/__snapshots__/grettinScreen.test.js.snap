// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proof <GrettingScreen/> should show the component correctly  1`] = `ShallowWrapper {}`;
